src/components/MessageList/MessageList.js

```diff
@@ -39,664 +45,370 @@ const ErrorNotification = styled.View`
+          setEditingState={setEditingState}
+          editing={editing}
+          threadList={threadList}
+          messageActions={messageActions}
+          updateMessage={updateMessage}
+          removeMessage={removeMessage}
+          retrySendMessage={retrySendMessage}
+          openThread={openThread}
+          emojiData={emojiData}
+          actionSheetStyles={actionSheetStyles}
+          AttachmentFileIcon={AttachmentFileIcon}
         />
       );
     }
   };
 
-  handleScroll = (event) => {
-    const yOffset = event.nativeEvent.contentOffset.y;
-    const removeNewMessageNotification = yOffset <= 0;
+  const handleScroll = (event) => {
+    const y = event.nativeEvent.contentOffset.y;
+    const removeNewMessageNotification = y <= 0;
     if (
-      !this.props.threadList &&
+      !threadList &&
       removeNewMessageNotification &&
-      this.props.channel.countUnread() > 0
+      channel.countUnread() > 0
     )
-      this.props.markRead();
+      markRead();
 
-    this.yOffset = yOffset;
-    this.setState((prevState) => ({
-      newMessagesNotification: removeNewMessageNotification
-        ? false
-        : prevState.newMessagesNotification,
-    }));
+    yOffset.current = y;
+    if (removeNewMessageNotification) {
+      setNewMessageNotification(false);
+    }
   };
 
-  renderEmptyState = () => {
-    const Indicator = this.props.EmptyStateIndicator;
-    return <Indicator listType='message' />;
+  const goToNewMessages = () => {
+    setNewMessageNotification(false);
+    flatListRef.current?.scrollToIndex({ index: 0 });
+    if (!threadList) markRead();
   };
 
-  render() {
-    const { t } = this.props;
-    const hasEventHistory =
-      this.props.eventHistory &&
-      Object.keys(this.props.eventHistory).length > 0;
-
-    // We can't provide ListEmptyComponent to FlatList when inverted flag is set.
-    // https://github.com/facebook/react-native/issues/21196
-    if (
-      this.props.messages &&
-      this.props.messages.length === 0 &&
-      !hasEventHistory &&
-      !this.props.threadList
-    ) {
-      return <View style={{ flex: 1 }}>{this.renderEmptyState()}</View>;
-    }
-
-    const TypingIndicator = this.props.TypingIndicator;
-    const HeaderComponent =
-      this.props.headerComponent || this.props.HeaderComponent;
-    const messagesWithDates = this.insertDates(this.props.messages);
-    const messageGroupStyles = this.getGroupStyles(messagesWithDates);
-    this.readData = this.getReadStates(messagesWithDates);
-    messagesWithDates.reverse();
-
-    const typing = Object.values(this.props.typing);
-    let showTypingIndicator;
-    if (
-      typing.length === 0 ||
-      (typing.length === 1 && typing[0].user.id === this.props.client.user.id)
-    ) {
-      showTypingIndicator = false;
-    } else {
-      showTypingIndicator = true;
-    }
+  const hasEventHistory = eventHistory && Object.keys(eventHistory).length > 0;
```

Comment:
We don't need EventHistory/eventHistory component anymore. So all the logic around it can be removed :)

Reply:
Awesome can do!

Reply:
I removed this and removed the addition of event histories to messages.